securedrop_client/gui/conversation/export/device.py

```diff
@@ -0,0 +1,135 @@
+import logging
+import os
+from typing import Optional
+
+from PyQt5.QtCore import QObject, QThread, pyqtSignal
+
+from securedrop_client.export import Export
+from securedrop_client.logic import Controller
+
+logger = logging.getLogger(__name__)
+
+
+class Device(QObject):
+
+    export_preflight_check_requested = pyqtSignal()
+    export_preflight_check_succeeded = pyqtSignal()
+    export_preflight_check_failed = pyqtSignal(object)
+
+    export_requested = pyqtSignal(list, str)
+    export_succeeded = pyqtSignal()
+    export_failed = pyqtSignal(object)
+    export_completed = pyqtSignal(list)
+
+    print_preflight_check_requested = pyqtSignal()
+    print_preflight_check_succeeded = pyqtSignal()
+    print_preflight_check_failed = pyqtSignal(object)
+
+    print_requested = pyqtSignal(list)
+    print_succeeded = pyqtSignal()
+    print_failed = pyqtSignal(object)
+
+    def __init__(
+        self, controller: Controller, export_service_thread: Optional[QThread] = None
+    ) -> None:
+        super().__init__()
+
+        self._controller = controller
+
```

Comment:
I like the idea of avoiding the dependency on the (big) controller class in the export and print dialogs if possible.  If that's the goal, I think we'd ideally avoid a dependency on the controller here too, otherwise we are adding a layer of abstraction between the dialogs and the controller, but not eliminating the dependency.  (@gonzalo-bulnes and I have chatted about a possible followup PR to accomplish this)

Reply:
I took a fresh look into what it would take to remove the controller dependency immediately. I think we were right and it's better in a separate PR, but indeed I'll be looking into creating that one rather sooner than later.